Add ops file to enable bbr

mhuangpivotal opened the pull request:
concourse/concourse#2332

Blocked by: https://github.com/concourse/concourse-bosh-release/pull/29

mhuangpivotal committed:
Add ops file to enable bbr

Signed-off-by: Mark Huang <mhuang@pivotal.io>
Co-authored-by: Bin Ju <bju@pivotal.io>

cirocosta commented on cluster/operations/bbr.yml:
```diff
@@ -0,0 +1,26 @@
+    name: bbr-atcdb
+    properties:
+      postgresql:
+        database: atc
```
Hey,

Do you think we need to make this configurable?

Thanks!

cirocosta commented on cluster/operations/bbr.yml:
```diff
@@ -0,0 +1,26 @@
+      postgresql:
+        database: atc
+        role:
+          name: concourse
```
Just like in the other comment, this *might* be important to be parametrizable. Wdyt?

thx!

cirocosta commented:
Hey,

As next steps, @mhuangpivotal and I were looking at some more possible testing that could be done regarding thre three possible ways of having the Database backed up:

- external
- cross-deployment
- intra-deployment

We've been thinking about having the Postgres properties propagated through links to the `bbr-atcdb` job.

thx!

taylorsilva committed:
Use bosh links. Get db info from the web job

Signed-off-by: Taylor Silva <tsilva@pivotal.io>

taylorsilva commented:
Decided to use the bosh links. PR here that updates the bosh release to use bosh links: https://github.com/concourse/concourse-bosh-release/pull/29

This should now allow us to do external, cross-deployment, and intra-deployment testing. These ops files would help us cover intra-deployment testing.

Changed the ops file to use an explicit linking from the web job instead of having any parameters set. Configure once, set everywhere!

taylorsilva commented on cluster/operations/backup-atc-colocated-web.yml:
```diff
@@ -0,0 +1,21 @@
+    name: bbr-atcdb
+    release: concourse
+    consumes:
+      concourse_db: {from: web}
```
An implicit link would've worked, so this `consumes` is technically not needed. If we removed them then bosh would just figure out the implicit linking. If we want less yaml then we can delete these lines. Thoughts?

cirocosta replied:
hmmmmm that's interesting - I personally prefer the explicitness that the `from: web` brings - seems like a nice clear way of seeing that the properties are "flowing" from `web` to the `bbr` job.  

Wdyt @YoussB @pivotal-bin-ju ?

taylorsilva replied:
We removed this because you need a matching provides for the consume.

cirocosta replied:
oh, interesting! Thanks!

taylorsilva committed:
Remove consumes bosh link

You can't use just a consumes, you need a provide AND consume.
If you only declare a consume and no provide then the deployment fails.

Signed-off-by: Taylor Silva <tsilva@pivotal.io>

cirocosta reviewed (approved):
SGTM 👍
